client/js/middleware/serialization/qbank/serializers/drag_and_drop.js

```diff
@@ -29,10 +29,10 @@ function serializeTargets(originalTarget, newTarget, fileIds) {
   })];
 }
 
-function serializeZones(originalZones, newZones, targetId) {
-  if (!newZones) { return null; }
+function serializeZones(originalZones, newZones, targetId, visible) {
+  if (!newZones && _.isNil(visible)) { return null; }
```

Comment:
Is this supposed to be checking that newZones is not undefined or that it is not empty? In either case it might be worth thinking about using `! _.isUndefined` or `!_.isEmpty` to make it more explicit.